File: check_stats.py
import os
from supabase import create_client
from dotenv import load_dotenv

load_dotenv()

url = os.environ.get("SUPABASE_URL") or os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY") or os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY") or os.environ.get("SUPABASE_KEY")

sb = create_client(url, key)

# Get distinct matchdays in player_scores
res_scores = sb.table("player_scores").select("matchday, total_points").execute()
scores = res_scores.data
print(f"Total player scores: {len(scores)}")
md_scores = {}
for s in scores:
    md = s.get("matchday")
    pts = s.get("total_points") or 0
    if md is not None:
        md_scores[md] = md_scores.get(md, 0) + (1 if pts > 0 else 0)
print("Matchdays with non-zero scores in player_scores:")
for md, count in sorted(md_scores.items()):
    print(f"  Matchday {md}: {count} player-scores > 0 points")

# Get fixtures count and status per matchday
res_fixtures = sb.table("fixtures").select("matchday, status, start_time").execute()
fixtures = res_fixtures.data
print("\nFixtures by matchday:")
md_fixtures = {}
for f in fixtures:
    md = f.get("matchday")
    status = f.get("status")
    start_time = f.get("start_time")
    if md is not None:
        if md not in md_fixtures:
            md_fixtures[md] = []
        md_fixtures[md].append((status, start_time))

for md, fixs in sorted(md_fixtures.items()):
    finished = sum(1 for status, _ in fixs if status == 'finished')
    print(f"  Matchday {md}: {len(fixs)} fixtures, {finished} finished. First start: {min(t for _, t in fixs if t)}")

res_status = sb.table("matchday_status").select("*").execute()
print("\nMatchday Status:", res_status.data)

File: check_stats.js
// Mock WebSocket global for older Node versions using Supabase
global.WebSocket = class {};

const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');

let envContent = '';
try {
  envContent = fs.readFileSync(path.join(__dirname, '.env'), 'utf8');
} catch (e) {
  console.error("Could not read .env file:", e.message);
}

const env = {};
envContent.split('\n').forEach(line => {
  const match = line.match(/^\s*([\w\.\-]+)\s*=\s*(.*)?\s*$/);
  if (match) {
    let key = match[1];
    let value = match[2] || '';
    if (value.startsWith('"') && value.endsWith('"')) {
      value = value.slice(1, -1);
    } else if (value.startsWith("'") && value.endsWith("'")) {
      value = value.slice(1, -1);
    }
    env[key] = value.trim();
  }
});

const url = env.SUPABASE_URL || process.env.SUPABASE_URL;
const key = env.SUPABASE_SERVICE_ROLE_KEY || env.SUPABASE_KEY || process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_KEY;

if (!url || !key) {
  console.error("Missing SUPABASE_URL or SUPABASE_KEY in env configuration", { url, key });
  process.exit(1);
}

const sb = createClient(url, key);

async function main() {
  console.log("Connecting to Supabase at", url);

  // 1. Get matchday_status
  const { data: statusData, error: statusErr } = await sb.from("matchday_status").select("*");
  if (statusErr) console.error("Error status:", statusErr);
  else console.log("\nMatchday Status:", statusData);

  // 2. Get distinct matchdays in player_scores
  const { data: scores, error: scoresErr } = await sb.from("player_scores").select("matchday, total_points");
  if (scoresErr) {
    console.error("Error scores:", scoresErr);
  } else {
    console.log(`\nTotal player scores: ${scores.length}`);
    const mdScores = {};
    for (const s of scores) {
      const md = s.matchday;
      const pts = s.total_points || 0;
      if (md != null) {
        mdScores[md] = (mdScores[md] || 0) + (pts > 0 ? 1 : 0);
      }
    }
    console.log("Matchdays with non-zero scores in player_scores:");
    for (const md of Object.keys(mdScores).sort((a,b) => a-b)) {
      console.log(`  Matchday ${md}: ${mdScores[md]} player-scores > 0 points`);
    }
  }

  // 3. Get fixtures status
  const { data: fixtures, error: fixErr } = await sb.from("fixtures").select("matchday, status, start_time");
  if (fixErr) {
    console.error("Error fixtures:", fixErr);
  } else {
    console.log("\nFixtures by matchday:");
    const mdFixtures = {};
    for (const f of fixtures) {
      const md = f.matchday;
      const status = f.status;
      const start_time = f.start_time;
      if (md != null) {
        if (!mdFixtures[md]) mdFixtures[md] = [];
        mdFixtures[md].push({ status, start_time });
      }
    }
    for (const md of Object.keys(mdFixtures).sort((a,b) => a-b)) {
      const fixs = mdFixtures[md];
      const finished = fixs.filter(f => f.status === 'finished').length;
      const started = fixs.filter(f => f.status === 'started' || f.status === 'finished').length;
      const firstStart = fixs.map(f => f.start_time).filter(Boolean).sort()[0];
      console.log(`  Matchday ${md}: ${fixs.length} fixtures, ${finished} finished, ${started} started. First start: ${firstStart}`);
    }
  }
}

main().catch(console.error);
